addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -168,8 +178,16 @@ func (wh *ClusterBootstrap) ValidateCreate(ctx context.Context, obj runtime.Obje
 	var allErrs field.ErrorList
 	// Iterating one by one because we need field info for getFieldPath, and some core packages can be nil
 
-	if err := wh.validateClusterBootstrapPackage(ctx, clusterBootstrap.Spec.CNI, clusterBootstrap.Namespace, getFieldPath("cni")); err != nil {
-		allErrs = append(allErrs, err)
+	// If  spec.cni == nil and clusterbootstrap is  annotated,  it is using unmanaged CNI, and we do not need to validate CNI
+	_, usingUnmanagedCNI := clusterBootstrap.Annotations[constants.UseUnmanagedCNI]
```

Comment:
ValidateUpdate is missing the same checks ?

Reply:
added checks to validateUpdate
done